types/Connection.d.ts

```diff
@@ -0,0 +1,212 @@
+
+      /** Gets the value of the option `key`. */
+      get(key: string): any;
+
+      /**
+         * Returns the [MongoDB driver `MongoClient`](http://mongodb.github.io/node-mongodb-native/3.5/api/MongoClient.html) instance
+         * that this connection uses to talk to MongoDB.
+         */
+      getClient(): mongodb.MongoClient;
+
+      /**
+         * The host name portion of the URI. If multiple hosts, such as a replica set,
+         * this will contain the first host name in the URI
+         */
+      readonly host: string;
+
+      /**
+         * A number identifier for this connection. Used for debugging when
+         * you have [multiple connections](/docs/connections.html#multiple_connections).
+         */
+      readonly id: number;
+
+      /**
+         * A [POJO](https://masteringjs.io/tutorials/fundamentals/pojo) containing
+         * a map from model names to models. Contains all models that have been
+         * added to this connection using [`Connection#model()`](/docs/api/connection.html#connection_Connection-model).
+         */
+      readonly models: Readonly<{ [index: string]: Model<any> }>;
+
+      /** Defines or retrieves a model. */
+      model<T, U, TQueryHelpers = {}>(
+            name: string,
+            schema?: Schema<T, U, TQueryHelpers>,
+            collection?: string,
+            options?: CompileModelOptions
+        ): U;
+      model<T>(name: string, schema?: Schema<T>, collection?: string, options?: CompileModelOptions): Model<T>;
+
+      /** Returns an array of model names created on this connection. */
+      modelNames(): Array<string>;
+
+      /** The name of the database this connection points to. */
+      readonly name: string;
+
+      /** Opens the connection with a URI using `MongoClient.connect()`. */
+      openUri(uri: string, options: ConnectOptions, callback: Callback<Connection>): Connection;
+      openUri(uri: string, callback: Callback<Connection>): Connection;
+      openUri(uri: string, options?: ConnectOptions): Promise<Connection>;
+
+      /** The password specified in the URI */
+      readonly pass: string;
+
+      /**
+         * The port portion of the URI. If multiple hosts, such as a replica set,
+         * this will contain the port from the first host name in the URI.
+         */
+      readonly port: number;
+
+      /** Declares a plugin executed on all schemas you pass to `conn.model()` */
+      plugin<S extends Schema = Schema, O = AnyObject>(fn: (schema: S, opts?: any) => void, opts?: O): Connection;
+
+      /** The plugins that will be applied to all models created on this connection. */
+      plugins: Array<any>;
+
+      /**
+         * Connection ready state
+         *
+         * - 0 = disconnected
+         * - 1 = connected
+         * - 2 = connecting
+         * - 3 = disconnecting
+         * - 99 = uninitialized
+         */
+      readonly readyState: ConnectionStates;
+
+      /** Sets the value of the option `key`. */
```

Comment:
here we had the information that set is the same as connection.options[key] = value.

The typings do not expose the options attribute, so I removed this remark here and the same for get